pkg/mesh_install/install.go

```diff
@@ -155,3 +157,80 @@ func (i *Installer) triggerInjectedSidecarReconciliation() {
 		}
 	}
 }
+
+type KindSortOrder []string
+
+var InstallOrder KindSortOrder = []string{
+	"PriorityClass",
+	"Namespace",
+	"NetworkPolicy",
+	"ResourceQuota",
+	"LimitRange",
+	"PodSecurityPolicy",
+	"PodDisruptionBudget",
+	"ServiceAccount",
+	"Secret",
+	"SecretList",
+	"ConfigMap",
+	"StorageClass",
+	"PersistentVolume",
+	"PersistentVolumeClaim",
+	"CustomResourceDefinition",
+	"SecurityContextConstraints",
+	"ClusterRole",
+	"ClusterRoleList",
+	"ClusterRoleBinding",
+	"ClusterRoleBindingList",
+	"Role",
+	"RoleList",
+	"RoleBinding",
+	"RoleBindingList",
+	"Service",
+	"DaemonSet",
+	"Pod",
+	"ReplicationController",
+	"ReplicaSet",
+	"Deployment",
+	"HorizontalPodAutoscaler",
+	"StatefulSet",
+	"Job",
+	"CronJob",
+	"IngressClass",
+	"Ingress",
+	"APIService",
+}
+
+func sortManifestsByKind(manifests []client.Object, ordering KindSortOrder) []client.Object {
+	sort.SliceStable(manifests, func(i, j int) bool {
+		return lessByKind(manifests[i], manifests[j], manifests[i].GetObjectKind().GroupVersionKind().Kind, manifests[j].GetObjectKind().GroupVersionKind().Kind, ordering)
+	})
+
+	return manifests
+}
+
+func lessByKind(a interface{}, b interface{}, kindA string, kindB string, o KindSortOrder) bool {
```

Comment:
Top level scoped variables and parameters should generally be the most descriptive in Go. Short variable names are intended for tightly scoped blocks, like `err` or `i` in a tiny for loop.
```suggestion
func lessByKind(a interface{}, b interface{}, kindA string, kindB string, order KindSortOrder) bool {
```

Reply:
Also do `a` and `b` need to be `interface{}`? Can they be `client.Object` since that's what gets passed in from the wrapper sort function?

Reply:
didnte even need that.  just used the string kind cause thats all we were using